asyncy/omg/ServiceContract.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+from .OmgExceptions import OmgMismatchedPropertyLengthError, \
+    OmgMissingKeyInBodyError, OmgPropertyKeyMissingTypeError, \
+    OmgPropertyValueMismatchError
+
+
+class ServiceContract:
+
+    @classmethod
+    def ensure_value_of_type(cls, _type, value):
+        """
+        Ensures that the value matches the expected type as listed on
+        https://microservice.guide/schema/actions/#arguments.
+
+        Supported types: int, float, string, list, map, boolean, or any
+        """
+        if _type == 'string':
+            cls.ensure_type(str, value)
+        if _type == 'int':
+            cls.ensure_type(int, value)
+        if _type == 'boolean':
+            cls.ensure_type(bool, value)
+        if _type == 'map':
+            cls.ensure_type(dict, value)
+        if _type == 'float':
+            cls.ensure_type(float, value)
+        if _type == 'list':
+            cls.ensure_type(list, value)
+        if _type == 'number':
+            cls.ensure_type_list([int, float], value, _type)
+        if _type == 'any':
+            return
+
+    @classmethod
+    def validate_output_properties(cls, output: dict, body, story,
+                                   line):
+        """
+        Verify all properties are contained in the return body
+        """
+        if (not body and output) or (not output and body):
```

Comment:
This is an XOR and can be written more concisely:

```suggestion
        if bool(body) ^ bool(output)
```

Reply:
done